app/controllers/oregon_digital/explore_collections_controller.rb

```diff
@@ -46,9 +46,29 @@ def my_collections
     end
 
     def total_viewable_items(id)
-      ActiveFedora::Base.where("member_of_collection_ids_ssim:#{id}").accessible_by(current_ability).count
+      Hyrax::SolrService.query("member_of_collection_ids_ssim:#{id}").length
     end
 
+    def osu_items(id)
+      Hyrax::SolrService.query("member_of_collection_ids_ssim:#{id} AND visibility_ssi:osu")
+    end
+
+    def uo_items(id)
+      Hyrax::SolrService.query("member_of_collection_ids_ssim:#{id} AND visibility_ssi:uo")
+    end
+
+    def osu_restricted?(id)
+      !osu_items(id).empty? && current_ability.cannot?(:show, osu_items(id).first, visibility: 'osu')
```

Comment:
As a note for later, we might run into a situation were the first item is not representative of all OSU/UO works. That's why I was hoping the `current_ability.cannot?(:show, ActiveFedora::Base, visibility: 'osu')` syntax would work.

We might have to do this to test all works, which may be slow: `current_ability.cannot?(:show,  {:any => osu_items(id)})`
Alternatively: 
```
dummy = ActiveFedora::Base.new
dummy.visibility = 'osu'
current_ability.cannot?(:show, dummy)
```